### ISSUE TITLE
Swap Normal stuck in Horizontal Position

### ISSUE THREAD

**Issue by Kubrith** [CreativeMD/LittleTiles#153]
I apologise if this isn't an informative post, or containing what you need, I'm unsure if there's anything I can collect for this.
I've also only picked Minecraft back up since ~1.4, so I haven't dealt with mods since then.

I'm running FTB Ultimate Reloaded, and I added a couple extra mods in, along with Optifine and shaders.

Version: 1.12.2
Mod Version Installed: LittleTiles_v1.5.0-pre130_mc1.12.2.jar

There's a few things I tried doing, but weren't like the videos linked on the page, such as no outline / overlays of tools and what no, but that not the main issue.

When trying to create a custom door I create a LittleRecipe of the door, and it was fine until I touched the "Swap Normal" button.
Since then, it's locked into it's horizontal position. I go into the recipe, click "Swap Normal" again, it shows that it changes to the vertical position, but when I hit save and go back into it nothing has changed.

Other changes such as the position of where it should hinge changes, but I can't seem to get the normal to change.

Again, apologies if this is vague, but if there's anything that could help explain or show it to you I can go and take a look.
If this is being caused by some conflict between mods or what no, then understandable.

Thanks

**Edit**: As a bit of a side not, the more I'm watching, compared to the tutorials by Mischief of Mice, I don't seem to be having the same experience at all, so I assume something is messing with the mod. 
When trying to place thing, I don't get a preview of the item or where it is roughly in the world either.

1. **When right clicking the recipe.**

![Swap Normal 01](https://user-images.githubusercontent.com/49419407/55764718-adc35580-5a64-11e9-9239-be320c3bdfcb.JPG)

2. **When "Swap Normal" has been clicked.**

![Swap Normal 02](https://user-images.githubusercontent.com/49419407/55764719-ae5bec00-5a64-11e9-9f12-615fc07dbbbf.JPG)

3. **After hitting save, and then right clicking again to inspect the recipe settings.**

![Swap Normal 03](https://user-images.githubusercontent.com/49419407/55764720-ae5bec00-5a64-11e9-8466-1eb47920cd7a.JPG)

**Comment by CreativeMD** [comment]
There are two different types of recipes. The default one and the advanced recipe. The advanced one allows you to place the structure directly. Maybe you are using the ordinary one, if that is the case you have to "craft" it first in the workbench.

About your main issue: There is no need to change the normal, actually it does not even work the intended way (unfortunately). I'm pretty sure what you are looking for is changing the axis. You can do this using the second of the right button below the preview.

Hope that fixes your issue, just reopen it if it still does not work.

**Comment by Kubrith** [comment]
Hey, sorry for the confusion, I got mixed up in what I thought the buttons were doing. I was hitting it thinking it was just spinning my view of the preview, and not actually changing it's orientation / which way it opened.

I see what you mean now, and was able to get it to swing the way I wanted.

However, I still do have still have an issue with previews. I was convinced it was something messing with it, a mod or a texture pack, but it looks like it's the shader I'm using.

I spent the last couple days trying to find an answer, or someone with a similar problem, and only tonight did I find your issue that you put up about previews and shader interaction: https://github.com/sp614x/optifine/issues/663

However, for me looking in specific direction doesn't work. Or if it does, very rarely. I took some screenshots as I have to use a 4G router as I don't actually have internet at home, otherwise I would have tried to take a video and upload it.

Previews are missing when I try to place thing, but if by some miracle they do pop up, and I hold my mouse still, they fade in and out constantly, so they aren't always there, even if I don't move.

I also have an issue that the guidelines / previews of tools and where you're going to be breaking (the overlay on the block when using the LittleHammer, and also when trying to drag shapes with the LittleChisel) don't appear once shaders are enabled also

**Doors / Item Preview Issues**: https://imgur.com/a/UZpCxrW

**LittleChisel No Previews**: https://imgur.com/a/MbXNyjY

The glass blocks made with the Chisel are invisible as well, but at this point it looks to be an issue with shaders similar, but maybe not exactly, to the one you raised before.
I realise that this is a "Turn off the shaders if you want to use the mod" solution, but I wasn't sure if this was worth raising again, or if I should have created a different issue as it isn't what I originally thought I had an issue with.

Apologies again if this is the wrong place, or I'm being an idiot.
